### PR TITLE
[DROOLS-7044] Drools 8 docs : Add a chapter regarding traditional rul…

### PR DESCRIPTION
**Pull request by tkobayas**
…e syntax

**JIRA**: 
https://issues.redhat.com/browse/DROOLS-7044

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>
</details>

### CODE REVIEW

**Comment by tkobayas** [comment]
Regarding OOPath performance, I'll address with the other JIRA https://issues.redhat.com/browse/DROOLS-7055

**Comment by tkobayas** [comment]
@mariofusco @kaldesai Please review and merge if it's okay, thanks!

File: drools-docs/src/modules/ROOT/pages/language-reference/_drl-rules.adoc [hunk 1]
```diff
@@ -51,6 +51,23 @@ A DRL file can contain single or multiple rules and queries, and can define reso
 
 Each rule must have a unique name within the rule unit. If you use the same rule name more than once in any DRL file in the unit, the rules fail to compile. Rule names generally must follow standard Java identifier conventions. However, you can enclose rule names with double quotation marks (`rule "rule name"`) to  prevent possible compilation errors, especially if you use spaces in rule names.
 
+[id="new-and-traditional-syntax_{context}"]
+== New and Traditional syntax
```
Comment (kaldesai) [inline drools-docs/src/modules/ROOT/pages/language-reference/_drl-rules.adoc]:
```suggestion
== New and traditional syntax
```

**Review by kaldesai — APPROVED** [review]
@tkobayas Your content looks good to me. Thank you!

File: drools-docs/src/modules/ROOT/pages/language-reference/_drl-rules.adoc (hunk 1 above)
Comment (tkobayas) [reply drools-docs/src/modules/ROOT/pages/language-reference/_drl-rules.adoc]: Applied the suggestion, thanks!

File: drools-docs/src/modules/ROOT/pages/language-reference/_drl-rules.adoc [hunk 2]
```diff
@@ -51,6 +51,23 @@ A DRL file can contain single or multiple rules and queries, and can define reso
...
+.Example rule for loan application with traditional syntax
+[source]
+----
+rule "Underage"
+  when
+    Applicant( applicantName : name, age < 21 )
+    $application : LoanApplication( applicant == applicantName )
+  then
+    $application.setApproved( false );
+    $application.setExplanation( "Underage" );
```
Comment (mariofusco) [inline drools-docs/src/modules/ROOT/pages/language-reference/_drl-rules.adoc:68]: Maybe it is not relevant for this specific example, but I believe that we should always show good practices in our code snippet, so I would put those setX in a `modify` block or add a `drools.update` statement at the end.

Comment (tkobayas) [reply drools-docs/src/modules/ROOT/pages/language-reference/_drl-rules.adoc:68]: Thanks. Applied the suggestion.

**Review by mariofusco — APPROVED** [review]

**Pull request merged**